frontend/lib/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -555,7 +559,17 @@ export function formatNumber(
       trimMantissa: true,
     })
   } else if (format === "localized") {
-    return new Intl.NumberFormat().format(value)
+    const locale = getLocaleLanguage()
+    try {
+      return new Intl.NumberFormat(locale).format(value)
```

Comment:
Directly using `navigator.languages` here might be even slightly better since it gives us a list of configured locales in the user browser ordered by priority. The number formatter iterates through those to match the first supported locale.